### ISSUE TITLE
EntisGLS error

### ISSUE THREAD

**Issue by fushi002** [marcussacana/StringReloads#17]
First of all, thank you for making a good tool.
I wanted to change the font of the translated game using this tool, and I was able to change it eventually.
But there is a little problem.
Most of the time, there is no problem, but there is a bug in which the game stops in very few sections.
This bug varies from person to person, and the interval that occurs is not constant.
But for some people, it happens unconditionally in certain sections.

![f02f211fc814695e72885d470df8c96e2878a30727981b6902e54689cbef260a](https://user-images.githubusercontent.com/88200209/221358334-b8d35810-ab18-4af4-8486-62fc9a3d075c.png)
![5d0eb0ccca8f506ea25b61653b4196ac7c7543dccba9db618efc2247a94f5fe6](https://user-images.githubusercontent.com/88200209/221358344-aa2b4cbe-af03-4614-a0a5-8a116b37c486.png)

I tried many things, and after tracking the cause, I found out that the dll file (library file) I attached to change the font was the cause.
In areas where bugs occur unconditionally, 
If you didn't use the wrapping function, you could see that there was no bug. 

We received the latest version of the 32-bit SRLWrapper file released here, and we modified the SRLWrapper.dll file to winmm.dll.

I wanted to create a log file for you, but when I changed the settings to create a log file, 
the game would freeze as soon as I started it, so I couldn't create a log file.

In this game, if you don't change the font, the string is truncated, so you must be able to change the font and font size.
And since the translation work has already been completed, I hope the bug will be fixed.

[link removed]

I think it's a difficult problem to predict just by the explanation, so I'm attaching the entire package. The font is in a separate font folder, so you can install it.

We may have to make a boring check for the test, but I attached a save file for the section where there are many errors.(Last file)

I ask for your help me. Thank you.

**Comment by marcussacana** [comment]
Did you tried run in both versions of the game, I mean retail and cracked?
I think this game is denuvo protected, so, may not be a bug at all but incompatibiliy with the Denuvo protection.

For logging, you may give a shot by enabling the `logfile=true` and `loglevel=DBG`

**Comment by fushi002** [comment]
> Did you tried run in both versions of the game, I mean retail and cracked? I think this game is denuvo protected, so, may not be a bug at all but incompatibiliy with the Denuvo protection.
> 
> For logging, you may give a shot by enabling the `logfile=true` and `loglevel=DBG`

It is not a cracked executable.
The game used a certified save file because the first time you authenticate a DVD, you don't need to authenticate it afterwards.
I was a little worried because the save file is stored in the user account folder.
Using the tool function you created, I solved this problem by forcing the save location to be a subfolder of the game.
Therefore, there is no need for cracks.


I think it generally works very well.
Does the protection you mentioned cause errors in even a fraction of the game?
In this case, only a small fraction of the errors occur, and it is a little suspicious that the error message being output is related to memory.
If you play the original game without using a tool, the motion is a little softer, whereas if you use the tool and run it, the game feels a little heavy.
I'm a little suspicious about that.

I am attaching the log file after creating it in the way you told me.

[log](https://drive.google.com/file/d/1u_SSm9dbLJ9p_gFcHkuF288FAZW5hIv9/view?usp=share_link)

Thank you.

**Comment by marcussacana** [comment]
Disable in the srl.ini the AutoInstall, you don't need that just for change the font.
And yes, it can be unstable due the protection, since the srl is modding the game memory and it will be able to detect it something strange...

But you seem to be using the heavy hook here:
```
[DBG] Hook "ststeady.exe->eslHeapAllocate" Compiled; Hook: 0x00000000038858B6; Bypass: 0x0000000006B40000
[DBG] Hook "ststeady.exe->eslHeapFree" Compiled; Hook: 0x0000000003885C26; Bypass: 0x0000000006B50000
```

**Comment by fushi002** [comment]
> 

Changed autoinstall to false in srl.ini file.
And inside the EntisGLSConfig.xml file,
<file path="$(CURRENT)\Patch"/>
Add a path,
<file name="GyeonggiT" path="Font\GyeonggiT.ttf"/>
<file name="GyeonggiT" path="Font\GyeonggiT.ttf"/>
I changed the font like this.
However, the game seems to print out the original font.

Just to be sure.
<display caption="スタディ§ステディ"
This part.
<display caption="スタディ§ステディTest"
I switched to , but the caption that the game outputs is still the original, スタディ§ステディ.

I tried another one.
I created a patch folder as a subfolder and tried adding the EntisGLSConfig.xml file to it, but the new font still does not apply

What else can I try?

Or please let me know if there is a good way to fix the heavy hooking.

**Comment by marcussacana** [comment]
If your problem is what I think it is, so, srl is basically incompatible with your game and there are nothing that can be done.

But you can do this:
1 - Download the oficial EntisGLS SDK 
2 - repack the file with the oficial SDK tool.
3 - Use FontMod instead srl for change the font
4 - remove the Srl from the game.

**Comment by fushi002** [comment]
> 

Thank you for your advice.

This game seems to use a bitmap font file exclusively for entisgls. (Extension name is .bmf)
In the entisglsdk configuration file, there is a program called bmfontmake, and I tried to change the font using it, but the font I wanted was still not printed.

Maybe it's because I don't know how to use that program.
Here's how you tried:
bmfontmake.exe /o msgfont.bmf /size 20 /font GyeonggiT.ttf
The bitmap font used by msgfont.bmf in this game.

Anyway, this didn't get the result that I wanted.

Fortunately, after a few tries, I was able to read the default Korean font (which is not a perfect solution), which is a bit hard to read.
![image](https://user-images.githubusercontent.com/88200209/221412673-490c35a6-622d-4a5f-83d5-bdba8916d002.png)


I hope this tool can be further developed in the future.
Thank you for your help.
Have a nice day!
